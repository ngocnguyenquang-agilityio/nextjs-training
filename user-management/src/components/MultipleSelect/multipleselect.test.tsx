import { fireEvent, render, screen } from '@testing-library/react';

// Components
import { MultipleSelect } from '.';

const mockOptions = [
  {
    name: 'HTML',
    logo: 'https://symbols.getvecta.com/stencil_25/37_html5.d4d8050235.svg',
    id: '1'
  },
  {
    name: 'CSS',
    logo: 'https://symbols.getvecta.com/stencil_25/14_css3.d930bfb832.svg',
    id: '2'
  },
  {
    name: 'Javascript',
    logo: 'https://symbols.getvecta.com/stencil_25/41_javascript.0ca26ec4ab.svg',
    id: '3'
  }
];

const mockSelectedOptions = [
  {
    name: 'Typescript',
    logo: 'https://symbols.getvecta.com/stencil_25/87_typescript.cb2d7326fa.svg',
    id: '4'
  },
  {
    name: 'React',
    logo: 'https://symbols.getvecta.com/stencil_25/72_react.76a8d36b4b.svg',
    id: '5'
  }
];

const defaultProps = {
  id: 'dropdown',
  label: 'Select options',
  options: mockOptions,
  selectedOptions: mockSelectedOptions,
  onSelect: jest.fn(),
  onRemove: jest.fn()
};

describe('Dropdown Component', () => {
  test('Should render correctly', () => {
    const comp = render(<MultipleSelect {...defaultProps} />);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<MultipleSelect {...defaultProps} />);
    expect(comp).toMatchSnapshot();
  });

  test('Should show options', async () => {
    render(<MultipleSelect {...defaultProps} />);
    const text = screen.getByLabelText('Select options');
    await fireEvent.click(text);
    const options = screen.getByTestId('options');
    expect(options).toBeInTheDocument();
  });

  test('Should render correct number of options', async () => {
    render(<MultipleSelect {...defaultProps} />);
    const text = screen.getByLabelText('Select options');
    await fireEvent.click(text);
    const options = screen.getByTestId('options');
    expect(options.children).toHaveLength(mockOptions.length);
  });

  test('Should call onCheck when select option', async () => {
    render(<MultipleSelect {...defaultProps} />);
    const text = screen.getByLabelText('Select options');
    await fireEvent.click(text);
    const selectedOption = screen.getByTestId('option-1');
    await fireEvent.click(selectedOption);
    expect(defaultProps.onSelect.mock.calls.length).toBe(1);
  });

  test('Should call onRemove when remove selected option', async () => {
    render(<MultipleSelect {...defaultProps} />);
    const text = screen.getByLabelText('Select options');
    await fireEvent.click(text);
    const selectedOption = screen.getByTestId('select-4');
    await fireEvent.click(selectedOption);
    expect(defaultProps.onRemove.mock.calls.length).toBe(1);
  });
});
