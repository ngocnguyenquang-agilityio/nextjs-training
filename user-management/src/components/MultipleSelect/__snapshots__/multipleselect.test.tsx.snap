// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="block mb-1.5 text-sm text-gray-400"
        for="dropdown"
      >
        Select options
      </label>
      <div
        class="mt-2 w-full border border-gray-500 rounded-lg focus:outline-2 focus:outline-blue-500 focus:border-none"
      >
        <div
          class="bg-transparent w-full p-2"
        >
          <div
            class="w-full py-2"
          >
            <span
              class="border rounded-full bg-blue-400 p-2 mx-0.5 text-sm text-center text-white"
              data-testid="select-4"
            >
              Typescript
            </span>
            <span
              class="border rounded-full bg-blue-400 p-2 mx-0.5 text-sm text-center text-white"
              data-testid="select-5"
            >
              React
            </span>
          </div>
        </div>
        <div
          class="relative w-full focus:outline-2 focus:outline-blue-500 focus:border-none"
        >
          <input
            class="bg-transparent block w-full border text-black text-sm font-md rounded-lg p-2.5 border-none outline-none focus:border-none focus:outline-none false false undefined"
            id="dropdown"
            placeholder="Select tech"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <label
      class="block mb-1.5 text-sm text-gray-400"
      for="dropdown"
    >
      Select options
    </label>
    <div
      class="mt-2 w-full border border-gray-500 rounded-lg focus:outline-2 focus:outline-blue-500 focus:border-none"
    >
      <div
        class="bg-transparent w-full p-2"
      >
        <div
          class="w-full py-2"
        >
          <span
            class="border rounded-full bg-blue-400 p-2 mx-0.5 text-sm text-center text-white"
            data-testid="select-4"
          >
            Typescript
          </span>
          <span
            class="border rounded-full bg-blue-400 p-2 mx-0.5 text-sm text-center text-white"
            data-testid="select-5"
          >
            React
          </span>
        </div>
      </div>
      <div
        class="relative w-full focus:outline-2 focus:outline-blue-500 focus:border-none"
      >
        <input
          class="bg-transparent block w-full border text-black text-sm font-md rounded-lg p-2.5 border-none outline-none focus:border-none focus:outline-none false false undefined"
          id="dropdown"
          placeholder="Select tech"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
